macros/sql/surrogate_key.sql

```diff
@@ -1,8 +1,28 @@
-{%- macro surrogate_key() -%}
+{%- macro surrogate_key(field_list) -%}
 
-{% set fields = [] %}
 
-{%- for field in varargs -%}
+{%- if varargs|length >= 1 %}
+
+{{ log("Warning: the `surrogate_key` macro now takes a single list argument instead of multiple string arguments. Multiple string arguments are longer supported and will be deprecated in a future release of dbt-utils.", info=True) }}
```

Comment:
Heya, check out this [nifty exception](https://docs.getdbt.com/docs/writing-code-in-dbt/jinja-context/exceptions/#warn)

This wasn't available to me when I last wrote code like this, but now you can do:

```suggestion
{{ exceptions.warn("Warning: the `surrogate_key` macro now takes a single list argument instead of multiple string arguments. Multiple string arguments are longer supported and will be deprecated in a future release of dbt-utils.") }}
```

Reply:
But then again, implementing that will necessitate a change to `require-dbt-version` to `>=0.15.1` in the `dbt_project.yml` file

Reply:
Let's do it! I'll also update my code to use this in a separate PR :)

Reply:
See #198

Reply:
Perfect. Makes sense! Have committed your suggested change.